[TRTLLM-5326] - Fix test coverage report generation

yiqingy0 opened the pull request:
# PR title

Please write the PR title by following template:

[JIRA ticket link/nvbug link/github issue link][fix/feat/doc/infra/...] \<summary of this PR\>

For example, assume I have a PR hope to support a new feature about cache manager of Jira TRTLLM-1000 ticket, it would be like

[TRTLLM-1000][feat] Support a new feature about cache manager

## Description

Please explain the issue and the solution in short.

## Test Coverage

<!--
Please list clearly what are the relevant test(s) that can safeguard the changes in the PR. This helps us to ensure we have sufficient test coverage for the PR.
-->

## GitHub Bot Help

`/bot [-h] ['run', 'kill', 'skip', 'reuse-pipeline'] ...`

Provide a user friendly way for developers to interact with a Jenkins server.

Run `/bot [-h|--help]` to print this help message.

See details below for each supported subcommand.

<details>

`run  [--disable-fail-fast --skip-test --stage-list "A10-1, xxx" --gpu-type "A30, H100_PCIe" --add-multi-gpu-test --only-multi-gpu-test --disable-multi-gpu-test --post-merge --extra-stage "H100_PCIe-[Post-Merge]-1, xxx"]`

Launch build/test pipelines. All previously running jobs will be killed.

`--disable-fail-fast ` *(OPTIONAL)* : Disable fail fast on build/tests/infra failures.

`--skip-test ` *(OPTIONAL)* : Skip all test stages, but still run build stages, package stages and sanity check stages. Note: Does **NOT** update GitHub check status.

`--stage-list "A10-1, xxx"` *(OPTIONAL)* : Only run the specified test stages. Examples: "A10-1, xxx". Note: Does **NOT** update GitHub check status.

`--gpu-type "A30, H100_PCIe"` *(OPTIONAL)* : Only run the test stages on the specified GPU types. Examples: "A30, H100_PCIe". Note: Does **NOT** update GitHub check status.

`--only-multi-gpu-test ` *(OPTIONAL)* : Only run the multi-GPU tests. Note: Does **NOT** update GitHub check status.

`--disable-multi-gpu-test ` *(OPTIONAL)* : Disable the multi-GPU tests. Note: Does **NOT** update GitHub check status.

`--add-multi-gpu-test ` *(OPTIONAL)* : Force run the multi-GPU tests. Will also run L0 pre-merge pipeline.

`--post-merge ` *(OPTIONAL)* : Run the L0 post-merge pipeline instead of the ordinary L0 pre-merge pipeline.

`--extra-stage "H100_PCIe-[Post-Merge]-1, xxx"` *(OPTIONAL)* : Run the ordinary L0 pre-merge pipeline and specified test stages. Examples: --extra-stage "H100_PCIe-[Post-Merge]-1, xxx".

### kill

`kill  `

Kill all running builds associated with pull request.

### skip

`skip --comment COMMENT `

Skip testing for latest commit on pull request. `--comment "Reason for skipping build/test"` is required. IMPORTANT NOTE: This is dangerous since lack of user care and validation can cause top of tree to break.

### reuse-pipeline

`reuse-pipeline `

Reuse a previous pipeline to validate current commit. This action will also kill all currently running builds associated with the pull request. IMPORTANT NOTE: This is dangerous since lack of user care and validation can cause top of tree to break.

</details>

ZhanruiSunCh committed:
chore: bump version to 0.20.0 (#4469)

Signed-off-by: ZhanruiSunCh <184402041+ZhanruiSunCh@users.noreply.github.com>

Shixiaowei02 committed:
fix: replace the image links in the blog (#4490)

Signed-off-by: Shixiaowei02 <39303645+Shixiaowei02@users.noreply.github.com>

tongyuantongyu committed:
fix: cleanup process tree for disaggregated test (#4116)

Signed-off-by: Yuan Tong <13075180+tongyuantongyu@users.noreply.github.com>

QiJune committed:
Cherry pick #4508 (#4512)

Chore: waive torch compile test cases of deepseek v3 lite (#4508)

waive torch compile test cases of deepseek v3 lite

Signed-off-by: junq <22017000+QiJune@users.noreply.github.com>

yuxianq committed:
Cherry pick https://github.com/NVIDIA/TensorRT-LLM/pull/4447 (#4517)

fix: skip weights defined in create_weights for pp. (#4447)

Signed-off-by: Yuxian Qiu <142763828+yuxianq@users.noreply.github.com>

kaiyux committed:
chore: Remove unused script (#4485)

* Remove unused script

Signed-off-by: Kaiyu Xie <26294424+kaiyux@users.noreply.github.com>

* Remove unused json file

Signed-off-by: Kaiyu Xie <26294424+kaiyux@users.noreply.github.com>

---------

Signed-off-by: Kaiyu Xie <26294424+kaiyux@users.noreply.github.com>

yuxianq committed:
chore: Deprecate autopp. (#4471)

Signed-off-by: Yuxian Qiu <142763828+yuxianq@users.noreply.github.com>

dcampora committed:
fix: Fix trtllm sampler beam width bug (#4507)

* Fix TRTLLMSampler.

Signed-off-by: Daniel Campora <961215+dcampora@users.noreply.github.com>

* Added type hint.

Signed-off-by: Daniel Campora <961215+dcampora@users.noreply.github.com>

---------

Signed-off-by: Daniel Campora <961215+dcampora@users.noreply.github.com>

crazydemo committed:
tests: update api change from decoder to sampler in test (#4479)

update

Signed-off-by: Ivy Zhang <25222398+crazydemo@users.noreply.github.com>

Funatiq committed:
docs: Add KV Cache Management documentation (#3908)

* docs: Add KV Cache Management documentation

* Introduced a new document detailing the hierarchy and event system for KV cache management, including definitions for Pool, Block, and Page.
* Updated the index.rst to include a reference to the new kv-cache-management.md file.

Signed-off-by: Robin Kobus <19427718+Funatiq@users.noreply.github.com>

* Update docs/source/advanced/kv-cache-management.md

Co-authored-by: Netanel Haber <58652339+netanel-haber@users.noreply.github.com>
Signed-off-by: Robin Kobus <19427718+Funatiq@users.noreply.github.com>

* Update KV Cache Pool Management

Signed-off-by: Robin Kobus <19427718+Funatiq@users.noreply.github.com>

* docs: Addcross-file links

Signed-off-by: Robin Kobus <19427718+Funatiq@users.noreply.github.com>

* docs: Clarify tokens_per_block

Signed-off-by: Robin Kobus <19427718+Funatiq@users.noreply.github.com>

* docs: Clarify acronyms

Signed-off-by: Robin Kobus <19427718+Funatiq@users.noreply.github.com>

---------

Signed-off-by: Robin Kobus <19427718+Funatiq@users.noreply.github.com>
Co-authored-by: Netanel Haber <58652339+netanel-haber@users.noreply.github.com>

ruodil committed:
test: add failed case in waive list and fix some test script issue for perf test (#4528)

add failed case in waive list and fix some test script issue

Signed-off-by: Ruodi <200874449+ruodil@users.noreply.github.com>

Tabrizian committed:
Add tritonrelease container (#4544)

Signed-off-by: Iman Tabrizian <10105175+tabrizian@users.noreply.github.com>

MartinMarciniszyn committed:
fix: [TRTLLM-325]WAR against security vulnerabilities in Python packages (#4539)

* fix: [TRTLLM-325]WAR against security vulnerabilities in Python packages

Signed-off-by: Martin Marciniszyn Mehringer <11665257+MartinMarciniszyn@users.noreply.github.com>

* Update docker images

Signed-off-by: Martin Marciniszyn Mehringer <11665257+MartinMarciniszyn@users.noreply.github.com>

---------

Signed-off-by: Martin Marciniszyn Mehringer <11665257+MartinMarciniszyn@users.noreply.github.com>

yiqingy0 commented:
/bot run

tensorrt-cicd commented:
[PR_Github #6138](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/6138/) [ run ] triggered by Bot

hyukn committed:
[5141290][5273694][5260696] fix: Fix mrope argument missing issue in the summary tasks for Qwen model. (#4432)

Fixed the mrope argument missing issue in the summary tasks for Qwen models.
And re-enabled the fixed tests.

Signed-off-by: Yukun He <23156053+hyukn@users.noreply.github.com>

tensorrt-cicd commented:
[PR_Github #6138](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/6138/) [ run ] completed with state `SUCCESS`
[/LLM/main/L0_MergeRequest_PR pipeline #4485](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/4485/) completed with status: 'SUCCESS'

ruodil committed:
test: waive hanging cases for perf test (#4563)

waive hanging cases

Signed-off-by: Ruodi <200874449+ruodil@users.noreply.github.com>

Funatiq committed:
[nvbugs/5274894] fix: Moving finished context requests to generation (#4576)

fix: Moving finished context requests to generation

- Unfinished chunked context requests appear at end of context requests vector.
- Replaced std::find_if with std::partition to find the correct position to move finished context requests to generation.

Signed-off-by: Robin Kobus <19427718+Funatiq@users.noreply.github.com>

hyukn committed:
[5234029][5226211] chore: Unwaive multimodal tests for Qwen model. (#4519)

Unwaive multimodal tests for Qwen models.

Signed-off-by: Yukun He <23156053+hyukn@users.noreply.github.com>

venkywonka committed:
test(perf): Extend the Llama-Nemotron-Nano-8B perf-integration-tests (pyt)  (#4407)

* extend pyt nano tests perf coverage

Signed-off-by: Venky <23023424+venkywonka@users.noreply.github.com>

* explicitly set maxnt for some cases

This is because the test harness default to no prefill chunking, that means the isl specified is the true context.
When explicitly unspecified in the test harness, the `maxnt` passed down to `trtllm-bench` is 2048.
This means trtllm-bench gets conflicting inputs when isl>2048 but maxnt=2048; hence overriding maxnt to be consistent with isl for such cases.

Signed-off-by: Venky <23023424+venkywonka@users.noreply.github.com>

---------

Signed-off-by: Venky <23023424+venkywonka@users.noreply.github.com>

yiqingy0 commented:
/bot run --post-merge

tensorrt-cicd commented:
[PR_Github #6199](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/6199/) [ run ] triggered by Bot

chzblych commented on jenkins/L0_MergeRequest.groovy:
```diff
@@ -714,8 +714,8 @@ def collectTestResults(pipeline, testFilter)
                 sh "cd cov && find . -type f"
-                sh "cd cov && coverage report"
-                sh "cd cov && coverage html -d test_coverage_html"
+                sh "cd cov && coverage report -i"   // -i: ignore errors. Ignore the error that the source code file cannot be found.
```
Could you add the comment and elaborate on why sometimes the source file cannot be found?

yiqingy0 replied:
This is the error log.
`[2025-05-21T08:27:43.152Z] /usr/local/lib/python3.12/dist-packages/coverage/report_core.py:116: CoverageWarning: Couldn't parse '/home/jenkins/agent/workspace/LLM/main/L0_Test-x86_64/llmVanilla/TensorRT-LLM/src/examples/config-3.py': No source for code: '/home/jenkins/agent/workspace/LLM/main/L0_Test-x86_64/llmVanilla/TensorRT-LLM/src/examples/config-3.py'. (couldnt-parse)

[2025-05-21T08:27:43.152Z]   coverage._warn(msg, slug="couldnt-parse")

[2025-05-21T08:27:43.152Z] /usr/local/lib/python3.12/dist-packages/coverage/report_core.py:116: CoverageWarning: Couldn't parse '/home/jenkins/agent/workspace/LLM/main/L0_Test-x86_64/llmVanilla/TensorRT-LLM/src/examples/config.py': No source for code: '/home/jenkins/agent/workspace/LLM/main/L0_Test-x86_64/llmVanilla/TensorRT-LLM/src/examples/config.py'. (couldnt-parse)

[2025-05-21T08:27:43.152Z]   coverage._warn(msg, slug="couldnt-parse")`

The triton stages, such as `[A30-Triton-Python-[Post-Merge]-1]`, seems create some files in testing process. I'm checking with Iman. I checked the files are not in the TensorRT-LLM repo. So I think we could ignore this error.

ruodil committed:
test: fix for perf sanity test and skip fp8 deepseek blackwell cases (#4598)

fix for sanity test and skip fp8 deepseek blackwell cases

Signed-off-by: Ruodi <200874449+ruodil@users.noreply.github.com>

hyukn committed:
[5180961] chore: Unwaive test for Qwen model. (#4524)

* Unwaive test for Qwen model.

Signed-off-by: Yukun He <23156053+hyukn@users.noreply.github.com>

* update.

Signed-off-by: Yukun He <23156053+hyukn@users.noreply.github.com>

---------

Signed-off-by: Yukun He <23156053+hyukn@users.noreply.github.com>

Barry-Delaney committed:
[https://nvbugs/5289907][fix] Restore per-channel pre-quant (#4545)

* Restore per-channel pre-quant

Signed-off-by: Barry Kang <43644113+Barry-Delaney@users.noreply.github.com>

* Update TRT test script

Signed-off-by: Barry Kang <43644113+Barry-Delaney@users.noreply.github.com>

* Fix pre-commit

Signed-off-by: Barry Kang <43644113+Barry-Delaney@users.noreply.github.com>

---------

Signed-off-by: Barry Kang <43644113+Barry-Delaney@users.noreply.github.com>

Barry-Delaney committed:
Update internal cutlass kernels commit id (#4619)

Signed-off-by: Barry Kang <43644113+Barry-Delaney@users.noreply.github.com>

stnie committed:
ci: waive testcase [NVBUG 5297821] (#4616)

Signed-off-by: Stefan Niebler <82932102+stnie@users.noreply.github.com>

Funatiq committed:
[nvbugs/5274894] fix: Sort requests for functional correctness and performance (#4608)

* Revert "[nvbugs/5274894] fix: Moving finished context requests to generation (#4576)"

This reverts commit d39bcb6b4079d138127b029bc718ff7a0397cb87.

Signed-off-by: Robin Kobus <19427718+Funatiq@users.noreply.github.com>

* fix: Sort requests for functional correctness and performance

- Moved sorting related logic to a dedicated function for better clarity and maintainability.
- Enhanced sorting logic to separate finished context requests from ongoing ones before sorting by Lora task ID.
- Updated function documentation to reflect the sorting behavior and its purpose.

Signed-off-by: Robin Kobus <19427718+Funatiq@users.noreply.github.com>

---------

Signed-off-by: Robin Kobus <19427718+Funatiq@users.noreply.github.com>

SimengLiu-nv committed:
[CI] Waive known errors with test TestDeepSeekV3Lite::test_fp8_block_scales_4gpus (#4627)

Signed-off-by: Simeng Liu <simengl@nvidia.com>

farazkh80 committed:
[TR[TLLM-4618][feat] Add remaining NVFP4 Nemotron Super 49B test on RTX6000 Pro (SM120) (#4548)

added nvfp4 nemotron for qa testing on RTX 6000

Signed-off-by: Faraz Khoubsirat <58580514+farazkh80@users.noreply.github.com>

moraxu committed:
[TRTLLM-4932] Add CLI accuracy tests for Llama-3_3-Nemotron-Super-49B-v1 and LLM API FP8 variant (#4375)

* Add CLI TestNemotronSuper acc tests

Signed-off-by: moraxu <mguzek@nvidia.com>

* Update mmlu.yaml

Signed-off-by: moraxu <mguzek@nvidia.com>

* Update yaml files

Signed-off-by: moraxu <mguzek@nvidia.com>

* Skip FP8 test in CLI

Signed-off-by: moraxu <mguzek@nvidia.com>

* Address reviews

Signed-off-by: moraxu <mguzek@nvidia.com>

* Address review comments

Signed-off-by: moraxu <mguzek@nvidia.com>

---------

Signed-off-by: moraxu <mguzek@nvidia.com>

moraxu committed:
[fix] Incorrect mocker argument for a CLI accuracy test in Llama-3.3-70B-Instruct (#4604)

Fix mocker argument

Signed-off-by: moraxu <mguzek@nvidia.com>

Tabrizian committed:
Add missing rcca folder (#4591)

Signed-off-by: Iman Tabrizian <10105175+tabrizian@users.noreply.github.com>

jinyangyuan-nvidia committed:
[fix] Fix Llama4 allgather error due to None tensor (#4511)

* [fix] Fix Llama4 allgather error due to None tensor

Signed-off-by: Jinyang Yuan <154768711+jinyangyuan-nvidia@users.noreply.github.com>

* Refactor modifications

Signed-off-by: Jinyang Yuan <154768711+jinyangyuan-nvidia@users.noreply.github.com>

* Minor modification

Signed-off-by: Jinyang Yuan <154768711+jinyangyuan-nvidia@users.noreply.github.com>

* Minor fix

Signed-off-by: Jinyang Yuan <154768711+jinyangyuan-nvidia@users.noreply.github.com>

---------

Signed-off-by: Jinyang Yuan <154768711+jinyangyuan-nvidia@users.noreply.github.com>

moraxu committed:
[TRTLLM-4932] Add QA accuracy tests for NIM-prioritized models (#4242)

* Add tests

Signed-off-by: moraxu <mguzek@nvidia.com>

* Add tests v2

Signed-off-by: moraxu <mguzek@nvidia.com>

* Add fixes

Signed-off-by: moraxu <mguzek@nvidia.com>

* Skip fp8 test for Ultra

Signed-off-by: moraxu <mguzek@nvidia.com>

* Add tests for Phi

Signed-off-by: moraxu <mguzek@nvidia.com>

* Skip tests for Phi

Signed-off-by: moraxu <mguzek@nvidia.com>

* Skip tests for Phi - fix

Signed-off-by: moraxu <mguzek@nvidia.com>

* Skip tests for Phi - comment out acc refs

Signed-off-by: moraxu <mguzek@nvidia.com>

* Add more test granularity

Signed-off-by: moraxu <mguzek@nvidia.com>

* Fix examples_test_list.txt

Signed-off-by: moraxu <mguzek@nvidia.com>

* Update test list file

Signed-off-by: moraxu <mguzek@nvidia.com>

* Update yaml files

Signed-off-by: moraxu <mguzek@nvidia.com>

* Address review comments

Signed-off-by: moraxu <mguzek@nvidia.com>

* Remove MMLU tests

Signed-off-by: moraxu <mguzek@nvidia.com>

* Add remaining models

Signed-off-by: moraxu <mguzek@nvidia.com>

---------

Signed-off-by: moraxu <mguzek@nvidia.com>

MartinMarciniszyn committed:
Update the description for NGC docker images (#4671)

Signed-off-by: Martin Marciniszyn Mehringer <11665257+MartinMarciniszyn@users.noreply.github.com>

chzblych committed:
[Test] - Correct waive the Slurm test stage (#4680)

Signed-off-by: Yanchao Lu <yanchaol@nvidia.com>

Superjomn committed:
fix[nvbug/5286515]: trtllm-llmapi-launch on single node single gpu  (#4529)

fix[nvbug/5286515]: trtllm-llmapi-launch on single node single gpu (#4428)

crazydemo committed:
tests: waive and unwaive QA test cases (#4644)

Signed-off-by: Ivy Zhang <25222398+crazydemo@users.noreply.github.com>
